### ISSUE TITLE
Should __wasi_clockid_t be a capability?

### ISSUE THREAD

**Issue by tlively** [WebAssembly/WASI#118]
The current [clock APIs](https://github.com/WebAssembly/WASI/blob/master/phases/unstable/docs/wasi_unstable_preview0.md#clock_res_get) do not require any capability to read the various clocks and therefore assume ambient authority to do so. Assuming clock information is a sensitive resource, this violates OCAP. It seems reasonable that timing information would indeed be sensitive, for example in contexts that need determinism or are sensitive to side channel timing attacks, so should these APIs be changed to take a clock capability instead of a clock ID?

**Comment by sunfishcode** [comment]
I agree. Similarly, it would be good to make `random_get` require a capability.

**Comment by programmerjake** [comment]
requiring a capability for both the time and random APIs will definitely be useful for cases where deterministic execution is desired.

**Comment by mdempsky** [comment]
> Similarly, it would be good to make `random_get` require a capability.

Why not just use `fd_read` with a file that always returns random data? E.g., reserve FD 3 as `STDRANDOM_FILENO`, or make it identifiable through `fd_prestat_get`.

**Comment by sunfishcode** [comment]
@mdempsky Following up with specific comments about `random_get` in https://github.com/WebAssembly/WASI/issues/185.
